src/main/test/java/core/basesyntax/WorkWithStore.java

```diff
@@ -0,0 +1,132 @@
+    private static StoreDao dao;
+
+    @BeforeAll
+    static void createHandleStrategy() {
+        actionHandlerMapTest = new HashMap<>();
+        actionHandlerMapTest.put(Task.ActionType.BALANCE, new BalanceActionHandler());
+        actionHandlerMapTest.put(Task.ActionType.SUPPLY, new SupplyActionHandler());
+        actionHandlerMapTest.put(Task.ActionType.PURCHASE, new PurchaseActionHandler());
+        actionHandlerMapTest.put(Task.ActionType.RETURN, new ReturnActionHandler());
+        actionStrategyTest = new ActionStrategyImpl(actionHandlerMapTest);
+        dao = new StoreDaoImpl();
+        handleTasksTest = new ProcessStoreServiceImpl(dao, actionStrategyTest);
+    }
+
+    @BeforeEach
+    void cleanStorage() {
+        dao.clean();
+    }
+
+    @Test
+    void isPathEmpty_inputFile() {
+        assertThrows(ValidationDataException.class, () -> new CreateTaskServiceImpl()
+                    .createTasks(path -> new ArrayList<>(), ""));
+    }
+
+    @Test
+    void isInterfaceFunction_null() {
+        assertThrows(ValidationDataException.class, () -> new CreateTaskServiceImpl()
+                .createTasks(null, "inputFile.csv"));
+    }
+
+    @Test
+    void isOneCorrectLine_ok() {
+        List<Task> tasks = new CreateTaskServiceImpl()
+                .createTasks(path -> readFileHelper(path), "testFiles/inputFileTestOneCrtLine.csv");
+        handleTasksTest.processAction(tasks);
+        Map<String, Integer> expected = new HashMap<>();
+        expected.put("banana", 30);
+        Map<String, Integer> actual = dao.getStorage();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void isManyCorrectLine_ok() {
+        List<Task> tasks = new CreateTaskServiceImpl()
+                .createTasks(path -> readFileHelper(path), "testFiles/inputFileTestCorrectLines.csv");
+        handleTasksTest.processAction(tasks);
+        Map<String, Integer> expected = new HashMap<>();
+        expected.put("banana", 140);
+        expected.put("apple", 80);
+        Map<String, Integer> actual = dao.getStorage();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void isWrongLine_notOk() {
+        assertThrows(ValidationDataException.class, () -> {
+            new CreateTaskServiceImpl()
+                    .createTasks(path -> readFileHelper(path), "testFiles/inputFileTestWrongLine.csv");
+        });
+    }
+
+    @Test
+    void isNumNegative_notOk() {
+        assertThrows(ValidationDataException.class, () -> {
+            new CreateTaskServiceImpl()
+                    .createTasks(path -> readFileHelper(path), "testFiles/inputFileTestNumNegative.csv");
+        });
+    }
+
+    @Test
+    void isFileEmpty_notOk() {
+        assertThrows(ValidationDataException.class, () -> {
+            new CreateTaskServiceImpl()
+                    .createTasks(path -> readFileHelper(path), "testFiles/inputFileTestEmpty.csv");
+        });
+    }
+
+    private List<String[]> readFileHelper(String path) {

```

Comment:
I didn't check tests. Because in that task it's not necessary. I have suggestion to move that method to separate class and also test that method.